LCM/dsc/engine/ConsistencyInvoker/ConsistencyInvoker.c

```diff
@@ -36,7 +36,12 @@ int main(int argc, char *argv[])
 
     if(strcmp(pythonCommand, PYTHON3_COMMAND) == 0)
     {
-        dscScriptPath = realloc(dscScriptPath, strlen(dscScriptPath) + strlen("/python3") + 1 );
+        char* tempDscScriptPath = realloc(dscScriptPath, strlen(dscScriptPath) + strlen("/python3") + 1 );
+
+        if(NULL == tempDscScriptPath) {
+            free(dscScriptPath);
+        }
```

Comment:
Did not understand

Reply:
Considering the previous statement:
`dscScriptPath = realloc(dscScriptPath, strlen(dscScriptPath) + strlen("/python3") + 1 );`
Let's assume dscScriptPath contains a non-null value to a memory location. Now, when realloc is unable to allocate memory, it returns null and dscScriptPath is assigned to null hence the previous memory location address which it was having is not freed and the address is lost. Hence, we are now first trying to realloc and get the new memory location first in a temporary variable, check if it is null so that the previous memory location can be freed.
Reference: https://stackoverflow.com/questions/27589846/dynamic-arrays-using-realloc-without-memory-leaks